internal/set/set_test.go

```diff
@@ -0,0 +1,80 @@
+package set_test
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/ops-agent/internal/set"
+	"gotest.tools/v3/assert"
+)
+
+func TestSliceToSet(t *testing.T) {
+	testSlice := []int{
+		1,
+		2,
+	}
+	testSet := set.SliceToSet(testSlice)
+	assert.Equal(t, len(testSet), len(testSlice))
+	for _, v := range testSlice {
+		assert.Assert(
+			t,
+			testSet.Contains(v),
+			"Set was missing key from original slice %s",
+			v,
+		)
+	}
+}
+
+func TestMapToSet(t *testing.T) {
+	testMap := map[string]string{
+		"key1": "",
+		"key2": "",
+		"key3": "",
+	}
+	testSet := set.MapToSet(testMap)
+	assert.Equal(t, len(testSet), len(testMap))
+	for k := range testMap {
+		assert.Assert(
+			t,
+			testSet.Contains(k),
+			"Set was missing key from original map %s",
+			k,
+		)
+	}
+}
+
+func TestAdd(t *testing.T) {
+	testSet := set.Set[string]{}
+	newKey := "new key!"
+	testSet.Add(newKey)
+	assert.Equal(t, len(testSet), 1)
+	assert.Assert(t, testSet.Contains(newKey))
+}
+
+func TestRemove(t *testing.T) {
+	key := "key"
+	testSet := set.Set[string]{
+		key: struct{}{},
+	}
+	testSet.Remove(key)
+	assert.Equal(t, len(testSet), 0)
```

Comment:
I would throw in an assert.Assert(t, !testSet.Contains(key)) here.

Reply:
Done in [8c8978f](https://github.com/GoogleCloudPlatform/ops-agent/pull/901/commits/8c8978ffc91e2e1e2dcc118c3017717a6d266f39)